sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -325,6 +328,36 @@ class JacksonParser(
         parseJsonToken[AnyRef](parser, dataType)(PartialFunction.empty[JsonToken, AnyRef])
   }
 
+  def makeKeyConverter(dataType: DataType): KeyConverter = dataType match {
+    case BooleanType => (key: String) => key.toBoolean
+    case ByteType => (key: String) => key.toByte
+    case ShortType => (key: String) => key.toShort
+    case IntegerType => (key: String) => key.toInt
+    case LongType => (key: String) => key.toLong
+    case FloatType => (key: String) => key match {
+      case "NaN" => Float.NaN
+      case "Infinity" => Float.PositiveInfinity
+      case "-Infinity" => Float.NegativeInfinity
```

Comment:
Is it case-sensitive?

Reply:
Good point. I got it from makeConverter, but in  Cast.processFloatingPointSpecialLiterals I think we have a more complete use case. I'm going to change it in makeConverter in other PR to avoid mix things.